homeassistant/components/leviosa_shades/config_flow.py

```diff
@@ -0,0 +1,152 @@
+"""Config flow for Leviosa shades Zone."""
+import logging
+
+from leviosapy import discover_leviosa_zones, validate_zone
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    BLIND_GROUPS,
+    DEVICE_FW_V,
+    DEVICE_MAC,
+    DOMAIN,
+    GROUP1_NAME,
+    GROUP2_NAME,
+    GROUP3_NAME,
+    GROUP4_NAME,
+    GROUP5_NAME,
+    GROUP6_NAME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): str,
+        vol.Required(GROUP1_NAME): str,
+        vol.Optional(GROUP2_NAME): str,
+        vol.Optional(GROUP3_NAME): str,
+        vol.Optional(GROUP4_NAME): str,
+        vol.Optional(GROUP5_NAME): str,
+        vol.Optional(GROUP6_NAME): str,
+    }
+)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class LeviosaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Manages the interaction with user when a Leviosa Zone needs to be setup."""
+
+    VERSION = 1
+    GROUPS = [
+        GROUP1_NAME,
+        GROUP2_NAME,
+        GROUP3_NAME,
+        GROUP4_NAME,
+        GROUP5_NAME,
+        GROUP6_NAME,
+    ]
+
+    def __init__(self):
+        """Initialize the Motion Blinds flow."""
+
+        self._host = None
+        self._host_uid = None
+        self._devices = {}
+
+    async def async_step_user(self, user_input=None):
+        """Perform discovery and present an input screen for each Zone discovered."""
+
+        _LOGGER.debug("Looking for Leviosa Zone HUBs")
+        self._devices = await discover_leviosa_zones()
+        _LOGGER.debug("Found %d Zones advertising on the network ", len(self._devices))
+        devs_to_be_removed = []
+        for dev_key in self._devices:
+            if self._host_already_configured(self._devices[dev_key]):
+                devs_to_be_removed.append(dev_key)
+        for dev in devs_to_be_removed:
+            self._devices.pop(dev)
+        _LOGGER.info("There are %d Zones can be included in Hass", len(self._devices))
+        Zones = list(self._devices.keys())
+        if len(Zones) == 1:
+            self._host = self._devices[Zones[0]]
+            self._host_uid = Zones[0]
+            return await self.async_step_connect()
+        if len(Zones) > 1:
+            return await self.async_step_select()
+
+        return self.async_abort(reason="no_new_devs")
+
+    async def async_step_select(self, user_input=None):
+        """Handle multiple motion gateways found."""
```

Comment:
The original design had it done for `one config entry to one gateway`. Because only one gateway can be bound at a time, the user can only click on one hub that he wants to bind at a time.

Reply:
I am not quite sure what you mean

Reply:
I misunderstood what happened in step_select, thought it was a multi select.